src/utils/reporting_workflow.js

```diff
@@ -218,18 +233,54 @@ function determineActionSequence(planDefinition) {
   return path;
 }
 
+function convertTimeToMs(value, unit) {
+  // assumption: we will never have to deal with units larger than weeks
+  // because those get complicated (ex, how many ms is a month?)
+
+  switch (unit) {
+    case 'ms':
+      return value;
+    case 's':
+      return value * 1000;
+    case 'min':
+      return value * 1000 * 60;
+    case 'h':
+      return value * 1000 * 60 * 60;
+    case 'd':
+      return value * 1000 * 60 * 60 * 24;
+    case 'wk':
+      return value * 1000 * 60 * 60 * 24 * 7;
+    default:
+      throw new Error(`unsupported unit ${unit}`);
+  }
+}
+
 async function executeWorkflow(context) {
   db.upsert(COLLECTION, context, c => c.id === context.id);
 
   const planDef = context.planDefinition;
   const ig = findProfile(planDef);
 
   while (context.currentActionSequenceStep < context.actionSequence.length) {
+    const currentActionId = context.actionSequence[context.currentActionSequenceStep];
+
+    if (currentActionId.startsWith('&&')) {
+      // delay the specified amount of time
+      const waitTime = Number(currentActionId.substring(2).split(' ')[0]); // split off the first && and pick out the time
+      context.currentActionSequenceStep++;
+
+      // TODO: do we even need a job scheduler?
```

Comment:
From some online literature it appears long setTimeouts aren't an issue. However, the only thing to consider is if we delay by a week and the server needs to be restarted the timeout thread would be lost. I'm also not sure how well it scales. For current purposes I think this is fine though